Source/WTF/wtf/HashFunctions.h

```
@@ -266,7 +266,12 @@ namespace WTF {
     template<HashableWithMemberFunction T> struct MemberBasedHash {
         static unsigned hash(const T& key) { return key.hash(); }
         static bool equal(const T& a, const T& b) { return a == b; }
-        static constexpr bool safeToCompareToEmptyOrDeleted = false;
+        static constexpr bool safeToCompareToEmptyOrDeleted = [] {
+            if constexpr (requires { T::safeToCompareToHashTableEmptyOrDeletedValue; })
```

Is there a way to write this that would fail to compile (rather than just returning either `true` or `false`) if this is a static function member or static data member of a type other than `bool`? I can't think of one but maybe there is a way.

---

You could probably do something like:

```
if constexpr (requires { T::safeToCompareToHashTableEmptyOrDeletedValue; }) {
    static_assert(std::same_as<decltype(T::safeToCompareToHashTableEmptyOrDeletedValue), bool>);
    return T::safeToCompareToHashTableEmptyOrDeletedValue;
} else
    return false 
```

---

`static_assert(std::same_as<decltype(T::safeToCompareToHashTableEmptyOrDeletedValue), const bool>);`